vortex-array/src/array/bool/mod.rs

```diff
@@ -252,4 +257,55 @@ mod tests {
         let scalar = scalar_at(&arr, 4).unwrap();
         assert!(scalar.is_null());
     }
+
+    #[test]
+    fn patch_sliced_bools() {
+        let arr = {
+            let mut builder = BooleanBufferBuilder::new(12);
+            builder.append(false);
+            builder.append_n(11, true);
+            BoolArray::from(builder.finish())
+        };
+        let sliced = slice(arr.clone(), 4, 12).unwrap();
+        let (values, offset) = sliced.clone().into_bool().unwrap().into_boolean_builder();
+        assert_eq!(offset, 4);
+        assert_eq!(values.as_slice(), &[254, 15]);
+
+        // patch the underlying array
+        let patches = Patches::new(
+            arr.len(),
+            PrimitiveArray::new(buffer![4u32], Validity::AllValid).into_array(),
+            BoolArray::from(BooleanBuffer::new_unset(1)).into_array(),
+        );
+        let arr = arr.patch(patches).unwrap();
+        let (values, offset) = arr.into_bool().unwrap().into_boolean_builder();
+        assert_eq!(offset, 0);
+        assert_eq!(values.as_slice(), &[238, 15]);
+
+        // the slice should be unchanged
+        let (values, offset) = sliced.into_bool().unwrap().into_boolean_builder();
+        assert_eq!(offset, 4);
+        assert_eq!(values.as_slice(), &[254, 15]); // unchanged
+    }
+
+    #[test]
+    #[ignore = "See https://github.com/spiraldb/vortex/issues/1774"]
```

Comment:
Is there any way to do "expected fail" so it breaks if we accidentally fix it?

Reply:
#[should_panic]?